File: client-apps/cli/internal/cli/clierr/exit_codes.go
package clierr

// Exit codes returned by the CLI. These follow a convention similar to
// BSD sysexits.h but tailored to the Stigmer domain. Scripts and CI
// pipelines can inspect $? to determine the error category without
// parsing stderr.
const (
	ExitSuccess    = 0
	ExitGeneral    = 1 // Internal, Aborted, Canceled, Unknown, non-gRPC errors
	ExitUsage      = 2 // InvalidArgument, FailedPrecondition, AlreadyExists
	ExitConnection = 3 // Unavailable, DeadlineExceeded, ResourceExhausted
	ExitAuth       = 4 // Unauthenticated, PermissionDenied
	ExitNotFound   = 5 // NotFound
)

File: client-apps/cli/src/errors/exit-codes.ts
// Exit codes returned by the CLI. The convention mirrors the Go CLI
// (internal/cli/clierr/exit_codes.go) so scripts and CI can branch on `$?`
// without parsing stderr. classify() maps gRPC/Connect status codes onto these.

export const ExitCode = {
  Success: 0,
  General: 1, // Internal, Aborted, remote Canceled, Unknown, non-gRPC errors
  Usage: 2, // InvalidArgument, FailedPrecondition, AlreadyExists
  Connection: 3, // Unavailable, DeadlineExceeded, ResourceExhausted
  Auth: 4, // Unauthenticated, PermissionDenied
  NotFound: 5, // NotFound
} as const;

export type ExitCode = (typeof ExitCode)[keyof typeof ExitCode];
